pkg/satellite/stream/stream.go

```diff
@@ -271,6 +289,21 @@ func (ss *satelliteStream) recvLoop() {
 			metrics.setStreamId(ss.streamId)
 		}
 
+		latestByteTime, _ := ptypes.Timestamp(timestampLastByteReceived)
+		if ss.enableAutoClose && ss.autoCloseTime.Sub(latestByteTime) < 1*time.Second && receivingBytes {
+			// Reset the auto close stream timer if still receiving bytes
+			if ss.autoCloseTimer != nil {
+				ss.autoCloseTimer.Reset(time.Second * ss.autoCloseDelay)
+			} else {
+				// Create the timer
+				ss.autoCloseTimer = time.AfterFunc(time.Second*ss.autoCloseDelay, func() {
```

Comment:
Actually I believe this is blocking. I'll look into finding an async method of this.